scripts/apps/authoring/authoring/controllers/AssociationController.ts

```diff
@@ -134,8 +136,15 @@ export function AssociationController(content, superdesk, renditions, notify) {
         }
 
         if (scope.field != null && scope.field.field_type === 'media' && updated != null && updated.order == null) {
-            // if the field is of type media-gallery, assign order to the item being added
-            updated['order'] = scope.currentIndex;
+            // get greatest order from current items(or -1 if there aren't any items) and add one
+            const nextOrder = (
+                Object.keys(scope.item.associations)
```

Comment:
I guess you could use here `getAssociationsByFieldId` here as well:
```ts
const items = getAssociationsByFieldId(...);
const nextOrder = items.length > 0 ? items[items.length - 1].order + 1 : 0;
```

Reply:
I've made it use `getAssociationsByFieldId`, but I am still reordering it in order not to depend on the sort order of `getAssociationsByFieldId`.